apps/server/src/modules/h5p-editor/service/libraryStorage.service.spec.ts

```diff
@@ -584,29 +580,54 @@ describe('LibraryStorage', () => {
 					fileStream.on('end', () => resolve(Buffer.concat(chunks).toString('utf-8')));
 				});
 
-				console.log(3);
-
 				expect(streamContents).toEqual(testFile.content);
 			});
 		});
 		describe('when getting file stats', () => {
-			it('should return parsed json', async () => {
+			it('should return file stats', async () => {
 				const { testingLib, testFile } = await setup();
 
+				const mockStats = {
+					LastModified: new Date(),
+					ContentLength: 15,
+				};
+
+				// @ts-expect-error partial mock
+				s3ClientAdapter.head.mockResolvedValueOnce(mockStats);
+
 				const stats = await storage.getFileStats(testingLib, testFile.name);
 
 				expect(stats).toMatchObject({
-					size: expect.any(Number),
-					// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
-					birthtime: expect.any(Object), // expect.any(Date) behaves incorrectly
+					size: mockStats.ContentLength,
+					birthtime: mockStats.LastModified,
 				});
 			});
 
-			it("should fail if file doesn't exist", async () => {
-				const { testingLib, testFile } = await setup(true, false);
+			it('should fail if filename is invalid', async () => {
+				const { testingLib } = await setup(true, false);
+
+				const getStats = storage.getFileStats(testingLib, '../invalid');
+				await expect(getStats).rejects.toThrowError('illegal-filename');
+			});
+
+			it('should throw NotFoundException if length or birthtime are undefined', async () => {
```

Comment:
1. "length" of what?
2. ...if length or birthtime **is** undefined...

Reply:
Changed wording to clarify